fission-web-server/library/Fission/Web/Server/Handler/User/ResendEmail.hs

```diff
@@ -0,0 +1,35 @@
+module Fission.Web.Server.Handler.User.ResendEmail (handler) where
+
+import           Servant
+
+import           Fission.Prelude
+
+import qualified Fission.Web.API.User.Email.Resend.Types      as API.Email
+
+import           Fission.Web.Server.Authorization.Types
+import qualified Fission.Web.Server.Challenge.Retriever.Class as Challenge
+import           Fission.Web.Server.Email.Class
+import           Fission.Web.Server.Email.Types
+import qualified Fission.Web.Server.Error                     as Web.Err
+import           Fission.Web.Server.Models
+
+
+handler ::
+  ( MonadThrow          m
+  , MonadLogger         m
+  , MonadEmail          m
+  , Challenge.Retriever m
+  )
+  => ServerT API.Email.Resend m
+
+handler Authorization { about = Entity _ User { userEmail = Nothing } } =
+  Web.Err.throw err422 { errBody = "There is no email associated with the user" }
```

Comment:
Pretty sure we discussed this one in Discord. Let's `Web.Err.throw $ NotFound Challenge`

Reply:
If you want a special message about the email-user relationship, we can also leverage the typesafe DB and have a `NotFound UserEmail`. Your choice 👍

Reply:
Hm. When I try to do something like `NotFound @UserEmail`, I get this error:
```
Data constructor ‘UserEmail’ cannot be used here
    (it comes from a data family instance)
```

I'm unfortunately not super type/data family literate, so I'm not sure why this doesn't work.

And I wouldn't want the error message to be something like "No email found" (which I'd define `instance Display (NotFound Email)` as), as I feel this is slightly confusing if there's not even an email in the request.